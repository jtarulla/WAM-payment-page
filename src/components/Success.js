import React, { Component } from 'react';
import page from '../assets/page-3.png';
import success from '../assets/success.png';
import loading from '../assets/loading.gif';

export class Payment extends Component {
	state = {
		loading: true,
		message: null
	};

	async componentDidMount() {
		const url = 'http://www.mocky.io/v2/5e3d41272d00003f7ed95c09';
		const response = await fetch(url);
		const data = await response.json();
		this.setState({ message: data, loading: false });
	}

	render() {
		const { message } = this.state;

		if (this.state.loading) {
		}

		return (
			<>
				<div className='page-number'>
					<img src={page} alt='page-number' />
				</div>
				<div className='container'>
					{this.state.loading ? (
						<div className='loading'>
							<img src={loading} alt='loading' />
						</div>
					) : (
						<div className='success'>
							<img src={success} alt='success' />
							<h2>{message.title}</h2>
							<p> {message.text} </p>
						</div>
					)}
				</div>
			</>
		);
	}
}

export default Payment;
