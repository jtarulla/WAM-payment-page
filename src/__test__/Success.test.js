import { shallow, mount } from 'enzyme';
import React from 'react';
import Success from '../components/Success';

describe('Success Component', () => {
	it('renders Success component without errors', () => {
		expect(shallow(<Success />)).toMatchSnapshot();
	});

	it('componentDidMount has been called', () => {
		jest.spyOn(Success.prototype, 'componentDidMount');
		const wrapper = mount(<Success />);
		expect(Success.prototype.componentDidMount.mock.calls.length).toBe(1);
	});

	it('allows us to set props', () => {
		const wrapper = mount(<Success loading={false} />);
		expect(wrapper.props().loading).toEqual(false);
		wrapper.setProps({ loading: true });
		expect(wrapper.props().loading).toEqual(true);
	});
});
