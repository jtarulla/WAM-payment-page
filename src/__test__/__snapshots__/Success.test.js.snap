// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success Component renders Success component without errors 1`] = `ShallowWrapper {}`;
